tool/mapeditor.re

```diff
@@ -1,18 +1,30 @@
 static mut MAP_EDITOR_STATE = MapEditorState {
     map_name: "",
     map: Tas::current_map(),
+    mode: MapEditorMode::None,
 };
 
 struct MapEditorState {
     map_name: string,
     map: RefunctMap,
+    mode: MapEditorMode,
+}
+
+enum MapEditorMode {
+    Play,
+    Edit,
+    None,
```

Comment:
This variant shouldn't be required as it's the same as `!CURRENT_COMPONENTS.contains(MAP_EDITOR_COMPONENT)`.